### ISSUE TITLE
DatePicker with initialValue considered empty

### ISSUE THREAD

**Issue by bananer** [vaadin/flow-components#6325]
### Description

Passing an initial value to the `DatePicker` constructur sets that initial value also as the "empty value". Consequently, when the value is unchanged (or changed back to the initial value) the DatePicker returns `true` from `isEmpty()`

### Expected outcome

DatePicker with initialValue returns fals from isEmpty()

### Minimal reproducible example

```kotlin
DatePicker(LocalDate.now()).isEmpty // returns true
DatePicker(LocalDate.now()).optionalValue // returns empty Optional
```

Workaround:

```kotlin
DatePicker().apply { value = LocalDate.now() }.isEmpty // returns false
```

### Steps to reproduce

see above

### Environment

Vaadin version(s): 24.3.12
OS: Debian 12

### Browsers

Issue is not browser related

**Comment by yuriy-fix** [comment]
I would agree that it could be confusing, but unfortunately at the moment it's a [documented behaviour](https://github.com/vaadin/flow/blob/00dd9ef5c368f1d8f138ca6ba91d20b1c2fe8b1d/flow-server/src/main/java/com/vaadin/flow/component/AbstractField.java#L137) and some users could rely on it. We would be able to change it with the upcoming major release. 
As a workaround you can set an initial value after initialisation of the field:
```
DatePicker datePicker = new DatePicker();
datePicker.setValue(LocalDate.now());
```

**Comment by Artur-** [comment]
Could you clarify how this is a documented behavior? The javadoc you refer to is for how `AbstractField` works but the `DatePicker` constructor only says
```
Convenience constructor to create a date picker with a pre-selected date in current UI locale format.
Params:
initialDate – the pre-selected date in the picker
See Also:
setValue(Object)
```

It says nothing about the value being used as the empty value - who would want that?
There is no need for `DatePicker` to call that particular `AbstractField` constructor
